snuba/datasets/groups.py

```diff
@@ -99,7 +101,7 @@ def __init__(self) -> None:
             ),
             right_node=TableJoinNode(
                 table_name=events_source.format_from(),
-                columns=events_source.get_columns(),
+                columns=events_source.get_columns() + ColumnSet([("time", DateTime())]),
```

Comment:
This appending seems strange but I assume it's necessary. What dis?

Reply:
Context:
Bucketed time columns came before we had a separation between a logical schema from a physical schema (clickhouse) for a dataset. To be fair they came even before datasets themselves. And we are still working on building that separation.
Since there was only one schema per dataset, that was the clickhouse schema, and bucketed columns were not part of the clickhouse table, they could not be part of the dataset schema (since they are not real columns). Thus they were just stored in the `time_group_columns` parameter when instantiating a dataset.
Now we have such separation, but we are still (well, we were before this PR) not using the logical schema anywhere. 
The logical schema in my opinion should be the client facing schema of the dataset, thus I think it should contain the bucketed columns since the client can use them and the client does not know they are any different from real clickhouse columns.
This appending is doing exactly this, it should be less weird on the other datasets: errors, events, outcomes, transactions and outcomes_raw. In those case the concatenation is assigned directly to the abstract_column_set (which is the logical schema today).
The groups dataset is "special", it is the only dataset that contains a join, here the logical schema still looks like (before we build entities) a cartesian product between the joined table. It is created through this JoinClause class, thus adding the bucketed columns happen here. The logical schema and the join structure should be decoupled relatively soon.

Also, the old `time_group_columns` and `time_parse_columns` attributes of the TimeseriesDataset class should disappear as soon as I am done with the AST.
Hope this helps, will be happy to provide more context if needed.